hexrd/ui/material_structure_editor.py

```diff
@@ -259,28 +259,9 @@ def update_material(self):
                 U_array.append(atom['U'])
 
         mat = self.material
-        mat._atominfo = np.array(info_array)
-        mat._atomtype = np.array(type_array)
-        mat._U = np.array(U_array)
-
-        old_unitcell = mat.unitcell
-
-        # Re-create the unit cell from scratch. This is easier to do
-        # right now than setting the variables and figuring out which
-        # properties need to be updated and in what order...
-        mat._unitcell = unitcell(mat._lparms, mat.sgnum, mat._atomtype,
-                                 mat._atominfo, mat._U, mat._dmin.getVal('nm'),
-                                 mat._beamEnergy.value, mat._sgsetting)
-
-        # Set the stiffness back on it if it existed
-        if hasattr(old_unitcell, 'stiffness'):
-            mat.unitcell.stiffness = old_unitcell.stiffness
-
-        # Compute the density
-        mat.unitcell.CalcDensity()
-
-        # Update the structure factor of the PData
-        mat.update_structure_factor()
+        mat.atominfo = np.array(info_array)
+        mat.atomtype = np.array(type_array)
+        mat.U = np.array(U_array)
```

Comment:
I think we can perform a check at this point to see if the values were changed or not. If those things are not checked, then the `unitcell` update is performed thrice, once for each set of values.

Reply:
Should we do that here, or in the hexrd setters?

[Here](https://github.com/HEXRD/hexrd/blob/74131509ceb5bb83c2b835d60ffa701b97618cf5/hexrd/unitcell.py#L1413-L1417), for instance, could maybe be:

```python
    @U.setter
    def U(self, Uarr):
        if hasattr(self, '_U') and np.array_equal(self._U, Uarr):
            return

        self._U = Uarr
        self.aniU = False
        if(Uarr.ndim > 1):
            self.aniU = True
            self.calcBetaij()
```

That would cover any other cases where we set `U` as well in hexrdgui.

Reply:
Yeah, you're right. `hexrd` would be a better place for that.